js/src/frameworks/openai.ts

```diff
@@ -132,18 +132,24 @@ export class OpenAIToolSet extends BaseComposioToolSet {
                 runId = id;
             }
 
-            if(!runId) {
+            if (!runId) {
                 continue;
             }
 
             // Handle the 'requires_action' event
             if (event.event === 'thread.run.requires_action') {
                 const toolOutputs = await this.handleAssistantMessage(event.data, entityId);
 
-                // Submit the tool outputs
-                await client.beta.threads.runs.submitToolOutputs(thread.id, runId, {
-                    tool_outputs: toolOutputs
-                });
+                // Check if the run is still in a state that accepts tool outputs
+                const currentRun = await client.beta.threads.runs.retrieve(thread.id, runId);
+                if (currentRun.status === "requires_action") {
```

Comment:
Chore: (Not important)Can we use constants for things like this? Easy to maintain and scope of error is also reduced